ilxutils/ilxutils/cli.py

```diff
@@ -0,0 +1,395 @@
+                return self.test_check(bp.format(label=label,
+                                                 ilx_id=entity['ilx'],
+                                                 rdf_type=entity['type']))
+            types_equal = self.is_equal(entity['type'], rdf_type) # type check
+            entity_super_ilx = entity['superclass']['ilx'] if entity['superclass'] else ''
+            supers_equal = self.is_equal(ex_super_ilx, _super_data['ilx']) # superclass check
+            if not types_equal or not supers_equal:
+                search_results = None # Although true the first time, if
+
+        # Found an entity that already exists with the given input and creator
+        if search_results:
+            entity = search_results[0]
+            bp = 'Entity {label} already exisits with ILX ID {ilx_id} and of type {rdf_type}'
+            return self.test_check(bp.format(label=label,
+                                             ilx_id=self.fix_ilx(entity['ilx']),
+                                             rdf_type=entity['type']))
+
+        # Generates ILX ID and does a validation check
+        url = self.base_path + 'ilx/add'
+        data = {'term': label,
+                'superclasses': [{
+                    'id': superclass_data['data']['id'],
+                    'ilx': superclass_data['data']['ilx']}],
+                'type': rdf_type}
+        data = superclasses_bug_fix(data)
+        output = self.post(url, data)['data']
+        if output.get('ilx'):
+            ilx_id = output['ilx']
+        else:
+            ilx_id = output['fragment']  # archetype of beta
+
+        # Uses generated ILX ID to make a formal row in the database
+        url = self.base_path + 'term/add'
+        data = {'label': label.replace('&#39;', "'").replace('&#34;', '"'),
+                'ilx': ilx_id,
+                'superclasses': [{
+                    'id': superclass_data['data']['id'],
+                    'ilx': superclass_data['data']['ilx']}],
+                'type': rdf_type}
+        data = superclasses_bug_fix(data)
+        if definition:
+            data.update({'definition':definition})
+        return self.post(url, data)
+
+
+def main():
+    doc = docopt(__doc__, version=VERSION)
+
+    client = Client()
+    if doc.get('triple'):
+        request = client.add_triple(subj = doc['<subject>'],
+                                    pred = doc['<predicate>'],
+                                    obj  = doc['<object>'])
+    elif doc.get('entity'):
+        request = client.add_entity(rdf_type   = doc['<rdf:type>'],
+                                    superclass = doc['<rdfs:sub*Of>'],
+                                    label      = doc['<rdfs:label>'],
+                                    definition = doc['<definition:>'])
+    elif doc.get('get'):
+        request, success = client.get_data_from_ilx(doc['<identifier>'])
+
+    client.log_info(request['data'])
```

Comment:
This line looks like it could fail if none of triple entity or get is selected. It may not because of the structure of what docopt outputs, but better to be defensive and add an else statement to the above with `response = {'data':''}` or something like that.